server.js

```diff
@@ -1,38 +1,267 @@
+const app = express();
+
+// Middleware --------------------------------
+app.use(cors());
+app.use(bodyParser.json());
+
+// Models ------------------------------------
+const Talk = new mongoose.model('Talk', {
+  description: String,
+  event: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Event',
+  },
+  main_speaker: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Speaker',
+  },
+  name: String,
+  speaker_occupation: String,
+  tags: Array,
+  url: String,
+});
+
+const Event = new mongoose.model('Event', {
+  name: String,
+});
+
+const Speaker = new mongoose.model('Speaker', {
+  name: String,
+});
+
+// To reset and populate database ------------
+if (process.env.RESET_DATABASE) {
+  const populateDatabase = async () => {
+    await Talk.deleteMany();
+    await Event.deleteMany();
+    await Speaker.deleteMany();
+
+    let events = [{}];
+    let speakers = [{}];
+
+    // EVENTS - Remove all dublicate events and create a new array
+    const allEvents = Array.from(new Set(tedData.map((talk) => talk.event)));
+    allEvents.forEach((item) => {
+      const newEvent = new Event({ name: item });
+      newEvent.save();
+      events.push(newEvent);
+    });
+
+    // SPEAKERS - Remove all dublicate events and create a new array
+    const allSpeakers = Array.from(
+      new Set(tedData.map((talk) => talk.main_speaker))
+    );
+    allSpeakers.forEach((item) => {
+      const newSpeaker = new Speaker({
+        name: item,
+      });
+      newSpeaker.save();
+      speakers.push(newSpeaker);
+    });
+
+    // ALL TALKS - including connected events and speakers
+    tedData.forEach((item) => {
+      item.tags = JSON.parse(item.tags);
+      const newTalk = new Talk({
+        ...item,
+        event: events.find((event) => event.name === item.event),
+        main_speaker: speakers.find(
+          (speaker) => speaker.name === item.main_speaker
+        ),
+      });
+      newTalk.save();
+    });
+  };
+  populateDatabase();
+}
+
+// Routes ----------------------------------------------------------------
 app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+  res.json(
+    'Welcome to TED-talks API - by Karin! Read full documentation here: 👉 https://github.com/karinnordkvist/karinnordkvist-Technigo-17-20-Mongo-API/blob/master/Documentation.md.'
+  );
+});
+
+// TALKS ---------------------------------------
+app.get('/talks', (req, res) => {
+  const queriedTalks = req.query.search;
+  const queryRegex = new RegExp(queriedTalks, 'i');
+  Talk.find({ name: queryRegex })
+    .populate('event')
+    .populate('main_speaker')
+    .then((data) => {
+      if (data.length === 0) {
+        res.status(400).json({ error: 'Invalid request' });
+      } else {
+        res.json(data);
+      }
+    });
+});
+
+// TALKS BY ID --------------------------------
+app.get('/talks/:id', (req, res) => {
+  Talk.findById(req.params.id, (err, data) => {
+    if (err) {
+      res.status(404).json({
```

Comment:
Nice way of handling errors/404-status codes all the way through your application 👍